packages/core/src/lib/utils/webauthn-client.js

```diff
@@ -0,0 +1,224 @@
+//@ts-check
+
+// Declare a SimpleWebAuthnBrowser variable as part of "window"
+
+/** @typedef {"authenticate"} WebAuthnAuthenticate */
+/** @typedef {"register"} WebAuthnRegister */
+/** @typedef {WebAuthnRegister | WebAuthnAuthenticate} WebAuthnOptionsAction */
+/**
+ * @template {WebAuthnOptionsAction} T
+ * @typedef {T extends WebAuthnAuthenticate ?
+ *  { options: import("@simplewebauthn/server/script/deps").PublicKeyCredentialRequestOptionsJSON; action: "authenticate" } :
+ *  T extends WebAuthnRegister ?
+ *  { options: import("@simplewebauthn/server/script/deps").PublicKeyCredentialCreationOptionsJSON; action: "register" } :
+ * never
+ * } WebAuthnOptionsReturn
+ */
+
+/**
+ * webauthnScript is the client-side script that handles the webauthn form
+ * 
+ * @param {string} baseURL is the base URL of the auth API
+ * @param {string} providerID is the ID of the webauthn provider
+ */
+export async function webauthnScript(baseURL, providerID) {
```

Comment:
Let's name it `authURL` to avoid confusion with `basePath`, an option that we recently added 😁